### ISSUE TITLE
change default policy and disable subscriptions for fresh installations

### ISSUE THREAD

**Issue by myrdd** [RequestPolicyContinued/requestpolicy#614]
I think the default (when RPC is installed) should be:

1. default policy: deny
2. no subscriptions

The reasons are

1. with "default deny" the browser is more secure
2. activated subscriptions (a) create requests for retreiving and updating the list and (b) create rules. One or both aspects might not be what users want.

**Advantages:** This emphasizes *user control*. Besides that, the default values mean a more secure browser.
**Disadvantages:** There is the risk that some users uninstall the addon instead of switching to `default deny`, which would be what they want.

We can lower the disadvantages by having a good setup page (https://github.com/RequestPolicyContinued/requestpolicy/issues/545) and maybe by showing a message in the menu where the user has to say „Okay“ to the settings.

I label this issue as a `discussion` for now.

**Comment by acrobat1** [comment]
That makes sense, it's after all what NoScript does by default -and thinking of it, if someone installs such extensions it will likely be someone careful about security and willing to take the extra steps to set his personal rules.

About this:
>Disadvantages: There is the risk that some users uninstall the addon instead of switching >to default deny, which would be what they want.

I don't understand, maybe you did mean `default allow` ?

Good point about subscription lists, is there any way to prevent them from modifying existing rules if they refer to the same origin/destination?

**Comment by myrdd** [comment]
> I don't understand, maybe you did mean `default allow` ?

Oh yes, you are right! I've corrected the first post.

> is there any way to prevent them from modifying existing rules for origin/destination that the user has already allowed/denied?

User rules always take precedence over subscription rules.

**Comment by nodiscc** [comment]
Ok with default deny and no subscriptions by default, provided RPContinued shows a good first run page to the user (described in #545).

**Comment by chrcoluk** [comment]
I agree with suggestion although it needs to allow internal stuff such as the speeddial plugin (that worked on original request policy but doesnt on this version with default deny).  This issue forced me to use default allow with subscriptions but I much prefer to use default deny.

**Comment by myrdd** [comment]
@nodiscc good. I agree.

> it needs to allow internal stuff such as the speeddial plugin (that worked on original request policy but doesnt on this version with default deny).

@chrcoluk could you please describe that problem (and how to reproduce it) in another issue? It has nothing to do with this issue nor with subscriptions at all.
Before creating the issue, when in "default deny" mode, please check RPC's request log to see if there's a blocked request by the Speed Dial addon.
